### PR TITLE
Master mdl 71636 columnsortorder peerreviewfixes

### PR DESCRIPTION
**Pull request by marcghaly**
Files that were altered or created are the following:

- admin/qbankplugins.php
- admin/settings/plugins.php
- lang/en/question.php
- lib/classes/plugininfo/qbank.php
- lib/db/services.php
- question/amd/build/sort_columns.min.js
- question/amd/build/sort_columns.min.js.map
- question/amd/src/sort_columns.js
- question/classes/bank/column_sort_order.php
- question/classes/bank/column_sort_order_manager.php
- question/classes/external/set_columnbank_order.php
- question/classes/local/bank/helper.php
- question/classes/local/bank/view.php
- question/edit.php
- question/sortcolumns.php
- question/templates/qbank_columnsortorder.mustache
- question/tests/behat/columnsortorder.feature
- question/tests/columnsortorder_test.php

Files that were altered, removed or moved when moving from `core` to `qbank`:

- admin/qbankplugins.php
- admin/settings/plugins.php
- lang/en/question.php
- lib/classes/plugininfo/qbank.php
- lib/db/services.php
- question/amd/build/sort_columns.min.js
- question/amd/build/sort_columns.min.js.map
- question/bank/columnsortorder/amd/build/sort_columns.min.js
- question/bank/columnsortorder/amd/build/sort_columns.min.js.map
- question/bank/columnsortorder/amd/src/sort_columns.js
- question/bank/columnsortorder/classes/column_sort_order.php
- question/bank/columnsortorder/classes/column_sort_order_manager.php
- question/bank/columnsortorder/classes/external/set_columnbank_order.php
- question/bank/columnsortorder/db/services.php
- question/bank/columnsortorder/lang/en/qbank_columnsortorder.php
- question/bank/columnsortorder/sortcolumns.php
- question/bank/columnsortorder/templates/columnsortorder.mustache
- question/bank/columnsortorder/tests/behat/columnsortorder.feature
- question/bank/columnsortorder/tests/columnsortorder_test.php
- question/bank/columnsortorder/version.php
- question/classes/external.php
- question/classes/local/bank/helper.php
- question/classes/local/bank/view.php
- question/edit.php

### CODE REVIEW

**Comment by marcghaly** [comment]
Hi @safatshahin, here is the PR for https://github.com/catalyst/moodle-MDL-70329/issues/62 (`71636`), I have left the commit where I moved from `core` to `qbank` , I will squash and rename properly to `qbank_columnsortorder` once review is done. Only point we have to check is the handling of disabling this qbank plugin / checking when it is enabled. 
Thank you !

**Comment by safatshahin** [comment]
Hi @marcghaly 
Can you please inform if this is in tracker review? if not we can do an internal review.
Cheers

**Comment by marcghaly** [comment]
Hi @safatshahin it is not yet in review, it would need an internal review first.
Thank you

**Comment by safatshahin** [comment]
Hi @marcghaly 
Can you please update the branches with the latest master so that they can be reviewed.
cheers!

File: question/classes/local/bank/view.php [hunk 1]
```diff
@@ -266,6 +269,10 @@ protected function get_question_bank_plugins(): array {
             $questionbankclasscolumns[$key] = $newpluginclasscolumn;
         }
 
+        if ($this->columnorder) {
```
[the remark this answers is not in the document]
Comment (marcghaly) [reply question/classes/local/bank/view.php:272]: Added a check if plugin is enabled.

File: question/classes/local/bank/view.php [hunk 2]
```diff
@@ -1301,4 +1308,11 @@ public function add_searchcondition($searchcondition): void {
         $this->searchconditions[] = $searchcondition;
     }
 
+    /**
+     * Gets visible columns.
+     * @return array $this->visiblecolumns Visible columns.
```
[the remark this answers is not in the document]
Comment (marcghaly) [reply question/classes/local/bank/view.php:1313]: Fixed.

**Comment by safatshahin** [comment]
Closing this as this is currently in tracker review.

**Pull request closed without merging**